whois-nrtm4/src/main/java/net/ripe/db/nrtm4/persist/NrtmVersionInformationDao.java

```diff
@@ -0,0 +1,92 @@
+package net.ripe.db.nrtm4.persist;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.util.Optional;
+import java.util.UUID;
+
+
+@Repository
+public class NrtmVersionInformationDao {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NrtmVersionInformationDao.class);
+    private final JdbcTemplate jdbcTemplate;
+    private final RowMapper<VersionInformation> rowMapper = (rs, rowNum) ->
+            new VersionInformation(
+                    rs.getLong(1),
+                    NrtmSourceHolder.valueOf(rs.getString(2)),
+                    rs.getLong(3),
+                    UUID.fromString(rs.getString(4)),
+                    NrtmDocumentType.valueOf(rs.getString(5))
+            );
+
+    @Autowired
+    public NrtmVersionInformationDao(@Qualifier("nrtmDataSource") final DataSource dataSource) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+    }
+
+    public Optional<VersionInformation> findLastVersion(final NrtmSource source) {
+        try {
+            return Optional.ofNullable(jdbcTemplate.queryForObject(
+                    "select vi.id, src.name, vi.version, vi.session_id, vi.type " +
```

Comment:
no need for optional.ofNullable if it throws empty result exception?

Reply:
true

Reply:
Not really you are sending the optional.empty in case of empty result exception

Reply:
It is a bit odd here to catch an exception that we don't want to throw.
I suggest 
* to catch DataAccessException exception and return an error in that case. 
* move that error (LOGGER.debug) to NotificationFileGenerationService behind lastVersion.isEmpty() 
I don't understand why we are returning "null" or empty value in case that we receive an error from the query.